android/ui/src/main/java/com/trafi/ui/Badge.kt

```diff
@@ -0,0 +1,258 @@
+                vector = vector,
+                text = badge?.text,
+                isHiddenLayoutFiller = false,
+                isStackedBadge = true
+            )
+        }
+        subbadge?.let {
+            Image(
+                subbadge,
+                modifier = Modifier
+                    .width(constants.subBadgeIconWidth)
+                    .height(constants.subBadgeIconHeight)
+                    .padding(end = 3.dp, bottom = 3.dp)
+                    .constrainAs(createRef()) {
+                        bottom.linkTo(parent.bottom)
+                        top.linkTo(parent.bottom)
+                        start.linkTo(parent.end)
+                        end.linkTo(parent.end)
+                    }
+            )
+        }
+    }
+}
+
+@Composable
+private fun BadgeFiller(
+    badgeType: BadgeType,
+    vector: ImageVector?,
+    text: String?,
+    isHiddenLayoutFiller: Boolean = false,
+    isStackedBadge: Boolean
+) {
+    Row(
+        modifier = Modifier.padding(
+            horizontal = getBadgeHorizontalPadding(badgeType, isStackedBadge),
+            vertical = getBadgeVerticalPadding(badgeType)
+        ),
+        horizontalArrangement = Arrangement.spacedBy(constants.spacer)
+    ) {
+        vector?.let {
+            Image(
+                vector,
+                colorFilter = ColorFilter.tint(AmbientContentColor.current),
+                modifier = Modifier.align(Alignment.CenterVertically)
+                    .width(constants.iconWidth)
+                    .height(if (isHiddenLayoutFiller) 0.dp else constants.iconHeight)
+            )
+        }
+        text?.let {
+            Text(
+                text = text,
+                style = getBadgeTextStyle(badgeType).copy(fontWeight = FontWeight.Bold),
+                modifier = Modifier.align(Alignment.CenterVertically)
+                    .padding(bottom = 1.dp)
+                    .wrapContentHeight(unbounded = true)
+            )
+        }
+    }
+}
+
+@Composable
+private fun getBadgeHeight(type: BadgeType): Dp {
+    return when (type) {
+        SMALL_BADGE -> constants.minHeightSmall
+        MEDIUM_BADGE -> constants.minHeightMedium
+    }
+}
+
+@Composable
+private fun getBadgeRounding(type: BadgeType): Dp {
+    return when (type) {
+        SMALL_BADGE -> constants.cornerRadiusSmall
+        MEDIUM_BADGE -> constants.cornerRadiusMedium
+    }
+}
+
+@Composable
+private fun getBadgeHorizontalPadding(type: BadgeType, isStackedBadge: Boolean): Dp {
+    return when (type) {
+        SMALL_BADGE -> constants.horizontalPaddingSmall + (constants.borderWidth * 2)
+        MEDIUM_BADGE -> constants.horizontalPaddingMedium + (constants.borderWidth * 2)
+    }
+}
+
+@Composable
+private fun getBadgeVerticalPadding(type: BadgeType): Dp {
+    return when (type) {
+        SMALL_BADGE -> 0.dp
+        MEDIUM_BADGE -> constants.verticalPaddingMedium
+    }
+}
+
+@Composable
+private fun getBadgeTextStyle(type: BadgeType): TextStyle {
```

Comment:
More idiomatic in Kotlin to write this (and most other derived values from `BadgeType`) as an extension property (or extension function if additional parameters are needed), i.e.

```kotlin
private val BadgeType.textStyle: TextStyle get() = //..
```